src/Type.h

```diff
@@ -923,15 +923,10 @@ extern Type* flatten_type(Type* t);
 // Returns the "maximum" of two type tags, in a type-promotion sense.
 extern TypeTag max_type(TypeTag t1, TypeTag t2);
 
-// Given two types, returns the "merge", in which promotable types
-// are promoted to the maximum of the two.  Returns nil (and generates
-// an error message) if the types are incompatible.
-TypePtr merge_types(const TypePtr& t1, const TypePtr& t2);
-
-// Given a list of expressions, returns a (ref'd) type reflecting
-// a merged type consistent across all of them, or nil if this
-// cannot be done.
-TypePtr merge_type_list(detail::ListExpr* elements);
+// Given a list of expressions, returns the maximal type consistent across
+// all of them, or nil if this cannot be done.  "Maximal" incorporates
+// notions of arithmetic coercion, but otherwise requires type-equivalence.
+TypePtr maximal_type(detail::ListExpr* elements);
```

Comment:
Out of my depth and probably just me, but this new name is rather close to the existing `max_type()`.

Ignore if there's no real other candidate or the distinction is clear enough.

Reply:
shallow_merge_type_list is probably too naive?

Reply:
It returns a single type, not a list of types (so I find the original name pretty strange!).  They're not a merge in the sense that one would expect, e.g., it does not coalesce compatible-but-not-identical records.  I'm open to other suggestions, but I think `shallow_merge_type_list` isn't as apt as `maximal_type`.